### PR TITLE
Support shortcodes (Fix #172)

### LINKED ISSUE #172: Make all compilers support shortcodes
It's a small change on each of:
- [x] orgmode
- [x] asciidoc
- [x] bbcode
- [x] commonmark
- [x] kramdown
- [x] markmin
- [x] mediawiki
- [x] misaka
- [x] mistune
- [x] odt
- [ ] rest_html5
- [x] textile
- [x] txt2tags
- [x] wiki
- [x] ~~wordpress_compiler~~

I have not checked if any of them already do.

### PR DESCRIPTION
**Pull request by ralsina**
Adding shortcode support to almost all plugin compilers. (Issue #172)

### CODE REVIEW

File: v7/markmin/markmin/__init__.py [hunk 1]
```diff
@@ -54,8 +54,16 @@ def compile_html(self, source, dest, is_two_file=True):
                 if not is_two_file:
                     spl = re.split('(\n\n|\r\n\r\n)', data, maxsplit=1)
                     data = spl[-1]
-                body = m2h.markmin2html(data, pretty_print=True)
-                out_f.write(body)
+                output = m2h.markmin2html(data, pretty_print=True)
+                output, shortcode_deps = self.site.apply_shortcodes(output, filename=source, with_dependencies=True, extra_context=dict(post=post))
```
Comment (Kwpolska) [inline v7/markmin/markmin/__init__.py]: You aren’t getting the post object anywhere.

Comment (ralsina) [inline v7/markmin/markmin/__init__.py]: Ack, I could not test the py2-only plugins. Fixing.

**Comment by Kwpolska** [comment]
Also, could we replace all this copy-pasted boilerplate with a single function in `nikola.utils`? (which would move this change to after a release)

File: v7/markmin/markmin/__init__.py (hunk 1 above)
Comment (ralsina) [inline v7/markmin/markmin/__init__.py]: Done

**Comment by ralsina** [comment]
Well, it would bw a couple of calls... I don't know. I just copy/pasted the code that works.

**Comment by Kwpolska** [comment]
Feel free to go either way and merge if you think copy-paste will be enough.

**Comment by ralsina** [comment]
Ack, will wait for tests and merge.

**Pull request merged**